supervisor/options.py

```diff
@@ -2060,6 +2096,51 @@ def make_group(self):
         from supervisor.process import FastCGIProcessGroup
         return FastCGIProcessGroup(self)
 
+
+class DirectoryConfig(object):
+    """Configuration for a required directory."""
+
+    def __init__(self, name, path, create, mode):
+        self.name = name
+        self.path = path
+        self.create = create
+        self.mode = mode
+
+    def __repr__(self):
+        return "DirectoryConfig({!r}, {!r}, {!r}, 0o{:o})".format(
+            self.name, self.path, self.create, self.mode
+        )
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, DirectoryConfig)
+            and self.name == other.name
+            and self.path == other.path
+            and self.create == other.create
+            and self.mode == other.mode
+        )
+
+    def check(self):
+        """Check if the directory exists, and potentially try to
+        create it. Return True if directory was created."""        
+        if os.path.exists(self.path):            
+            if not os.path.isdir(self.path):             
+                raise ValueError(
+                    "required directory (%s) %s is not a directory"
+                    % (self.name, self.path)
+                )
+        else:            
+            if self.create:                            
```

Comment:
shouldn't the syntax support chowning the directory? if the directory is supposed to be created in /tmp then it will suffice but otherwise it may result with a permission error. had this be created by root then the config could also have a syntax for chown'ing the created directory

Reply:
Maybe is should. There was a comment at the end of https://github.com/Supervisor/supervisor/issues/120 that suggested it didn't need to change the ownership.